lib/ui/settings/darkmodesettings.js

```diff
@@ -0,0 +1,75 @@
+/*
+ * dark mode settings
+ *
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import {SplitButton, MenuItem} from 'react-bootstrap';
+
+import InstrumentPanel from '../instrumentpanel';
+import {
+  removeKeysByName,
+  retrieveDarkMode as localstorageRetrieveDarkMode,
+  saveDarkMode as localstorageSaveDarkMode
+  } from '../../util/localstorage';
+const menuSettings = {
+  mainbar: {title: 'By main bar icon', help: 'The dark mode is set by icon in main bar'},
+  os: {title: 'By OS settings', help: 'The dark mode follows your OS settings'},
+  skpath: {title: 'By Signal K Mode', help: 'The dark mode follows the value of Signal K path: /vessels/self/environment/mode'}
+}
+
+export const darkModeSetByMainbar = "mainbar";
+export const darkModeSetByOs = "os";
+export const darkModeSetBySkPath = "skpath";
+export const darkModeLight = "light";
+export const darkModeDark = "dark";
```

Comment:
- [X] const changed to :
-  CS_BY_MAINBAR
-  CS_BY_OS
-  CS_BY_SKPATH
-  CS_LIGHT
-  CS_DARK

- [X] renaming darkMode => colorScheme

- [X] For the design of the setting panel:
I chose to replace the tabs with a drop-down list because on a large screen, the tabs are fine, but on a smartphone, it takes up half the screen and it's really not very beautiful.
I adjusted the layout a little bit with more explicit texts and avoided repetitions.

What do you think of that?
![image](https://user-images.githubusercontent.com/31212553/68043016-997ec880-fcd4-11e9-8878-1d8c5c53cf07.png)

Reply:
Ok for me.

We could also render (React) or show (CSS) conditionally drop-down or tabs depending on the screen size, but we can go with this. Settings after all is not something you look at every day.